main.go

```diff
@@ -133,10 +152,37 @@ func (s *admissionWebhookServer) unmarshal(in *admissionv1.AdmissionRequest) (p
 		return "", nil, nil
 	}
 	p = path.Join("/", p)
-	if metaPtr.Labels == nil {
-		metaPtr.Labels = make(map[string]string)
+	if podMetaPtr.Labels == nil {
+		podMetaPtr.Labels = make(map[string]string)
+	}
+	// Annotations shouldn't be applied second time.
+	if isReplicaOwnedByDeployment(in.Kind.Kind, metaPtr) {
+		return "", nil, nil
+	}
```

Comment:
@anastasia-malysheva Could you say more about this?

Reply:
@denis-tingaikin, replicaSet inherits annotations from the deployment it is owned by. As soon as deployment already has annotations and they've been applied before replicaset was created, replicaset tries to apply same annotations again and it causes errors during installation. That's why this check is here.